File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mariadb/Group.java
package dev.skyherobrine.service.models.mariadb;

import com.fasterxml.jackson.annotation.JsonFormat;
import dev.skyherobrine.service.enums.GroupStatus;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity @Table(name = "groups")
@Getter @Setter
@NoArgsConstructor @RequiredArgsConstructor
public class Group {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "group_id", length = 15, nullable = false, unique = true)
    private String groupId;
    @Column(name = "group_name", length = 100, nullable = false) @NonNull
    private String groupName;
    @Column(name = "group_description", length = 1000)
    private String groupDescription;
    @Column(name = "allow_join", nullable = false)
    private Boolean allowJoin;
    @Column(name = "allow_invite", nullable = false)
    private Boolean allowInvite;
    @Column(name = "allow_notification", nullable = false)
    private Boolean allowNotification;
    @Column(name = "allow_chat", nullable = false)
    private Boolean allowChat;
    @Column(name = "wait_for_response", nullable = false)
    private Boolean waitForResponse;
    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    @Enumerated(EnumType.ORDINAL)
    @Column(nullable = false)
    private GroupStatus status;

    @PrePersist
    public void onPersist() {
        createdAt = updatedAt = LocalDateTime.now();
        allowJoin = allowInvite = allowNotification = allowChat = true;
        waitForResponse = false;
        status = GroupStatus.ACTIVE;
    }

    @PreUpdate
    public void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: healthcare_app_server/chat_service/models/mariadb/group.js
const group_status = {
    "ACTIVE": 0,
    "BLOCKED": 1,
    "DELETED": 2
}

class Group {
    constructor(id, groupId, groupName) {
        this._groupId = groupId;
        this._groupName = groupName;
        this._group_description = null;
        this._allow_join = this._allow_invite = this._allow_notification = this._allow_chat = true;
        this._wait_for_response = false;
        this._created_at = this._updated_at = new Date();
        this._status = 0;
        this._id = id;
        this._groupId = groupId;
        this._groupName = groupName;
        this._status = group_status.ACTIVE;
    }

    get id() {
        return this._id;
    }

    get groupId() {
        return this._groupId;
    }

    get groupName() {
        return this._groupName;
    }

    get group_description() {
        return this._group_description;
    }

    get allow_join() {
        return this._allow_join;
    }

    get allow_invite() {
        return this._allow_invite;
    }

    get allow_notification() {
        return this._allow_notification;
    }

    get allow_chat() {
        return this._allow_chat;
    }

    get wait_for_response() {
        return this._wait_for_response;
    }

    get created_at() {
        return this._created_at;
    }

    get updated_at() {
        return this._updated_at;
    }

    get status() {
        return this._status;
    }
}